### PR TITLE
Plugin v2

### PR DESCRIPTION
**Pull request by beygorghor**

### CODE REVIEW

File: hat/assets/js/apps/Iaso/constants/menu.js [hunk 1]
```diff
@@ -162,4 +163,10 @@ if (PLUGIN_POLIO_ENABLED === 'True') {
     });
 }
 
-export default menuItems;
+const plugins = getPlugins();
+let pluginsMenu = [];
+plugins.forEach(plugin => {
+    pluginsMenu = [...pluginsMenu, ...plugin.menu];
+});
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/constants/menu.js:170]: I would replace lines 166-170 with 
```javascript
const pluginsMenu = getPlugins().map(plugin=>plugin.menu).flat()
```
If we don't have to keep `menuItems` and `pluginsMenu` as separate variables (eg for readability), we could even spread it directly in `menuItems` (around line 147)

File: hat/assets/js/apps/Iaso/constants/routes.js [hunk 2]
```diff
@@ -496,28 +496,38 @@ export const page500 = {
...
+        dataSourcesPath,
+        tasksPath,
+        devicesPath,
+        groupsPath,
+        orgUnitTypesPath,
+        polioPath,
+        page401,
+        page404,
+        page500,
+    ];
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/constants/routes.js:524]: Same suggestion as in menu.js:
```javascript
    // routes
    page500,
    ...getPlugins().map(plugin=>plugin.routes).flat(),
];
```

Then we dont have to export `routeConfigs` as a function

File: hat/assets/js/apps/Iaso/domains/users/utils.js [hunk 3]
```diff
@@ -77,6 +77,6 @@ export const getFirstAllowedUrl = (rootPermission, user) => {
             newRoot = p;
         }
     });
-    const newPath = routeConfigs.find(p => p.permission === newRoot);
+    const newPath = routeConfigs().find(p => p.permission === newRoot);
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/utils.js:80]: cf routes.js

File: hat/assets/js/apps/Iaso/index.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ import { addRoutes } from './routing/redirections';
 import * as zoomBar from './components/leaflet/zoom-bar'; // don't delete - needed to override leaflet zoombar
 
 export default function iasoApp(element) {
-    const baseRoutes = routeConfigs.map(routeConfig => (
+    const baseRoutes = routeConfigs().map(routeConfig => (
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/index.js:16]: cf routes.js

File: hat/webpack.dev.js [hunk 5]
```diff
@@ -7,6 +7,9 @@ var BundleTracker = require('webpack-bundle-tracker');
 var LOCALE = 'fr';
 var WEBPACK_URL = 'http://localhost:3000';
 
+const pluginsString = process.env.PLUGINS || '';
+const plugins = pluginsString.split(',');
+
```
Comment (quang-le) [inline hat/webpack.dev.js:12]: if you want a one-liner:
`const plugins = process.env.PLUGINS?.split(',')??[];`

File: hat/webpack.prod.js [hunk 6]
```diff
@@ -6,6 +6,8 @@ const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 // remember to switch in webpack.dev.js and
 // django settings as well
 var LOCALE = 'fr';
+const pluginsString = process.env.PLUGINS || '';
+const plugins = pluginsString.split(',');
```
Comment (quang-le) [inline hat/webpack.prod.js:10]: cf webpack.dev.js

File: hat/webpack.dev.js (hunk 5 above)
Comment (quang-le) [reply hat/webpack.dev.js:12]: Not sure it's super readable though

Comment (beygorghor) [reply hat/webpack.dev.js:12]: yes i will leave it like that

**Review by quang-le — APPROVED** [review]

**Review by olethanh — APPROVED** [review]

**Pull request merged**